app/services/distribution_reminder.rb

```diff
@@ -1,11 +1,12 @@
 # This job notifies a Partner that they have a distribution scheduled to be sent in 24 hours
-class DistributionReminderJob < ApplicationJob
-  def perform(dist_id)
-    distribution = Distribution.find(dist_id)
+class DistributionReminder
+  def self.perform(dist_id)
+    distribution = Distribution.find_by(id: dist_id)
 
     # NOTE: This is being also checked in the DistributionMailer itself
-    return if distribution.past? || !distribution.partner.send_reminders
+    return if distribution.nil? || distribution.past? || !distribution.partner.send_reminders
```

Comment:
I wonder if we can combine these three conditions into a method that we check?

Reply:
Good idea! I'll put them into a private method.